.github/workflows/mvn-verify-check.yml

```
@@ -54,7 +54,18 @@ jobs:
           cache: 'maven'
       - run: |
           set -x
-          mvn initialize -am
+          max_retry=3; delay=30; i=1
+          while true; do
+            mvn -Dmaven.wagon.http.retryHandler.count=3 \
+              -Dmaven.wagon.httpconnectionManager.ttlSeconds=30 \
+              -Daether.connector.http.connectionMaxTtl=30 initialize -am && break || {
```

it is failing because it should be
```suggestion
              -Daether.connector.http.connectionMaxTtl=30 initialize -pl sql-plugin-api -am && break || {
```

---

Now we need to make sure that the `cache-dependencies` job completes before the matrix kicks in. The easiest way is to make it a dependency of ` get-shim-versions-from-dist`. To this end, add `needs: cache-dependencies` to the `get-shim-versions-from-dist` job

```yaml
  get-shim-versions-from-dist:
    needs: cache-dependencies
    ...
```

to

---

The previous tests failed since I didn't specify a project in `initialize` command. Some dependencies cannot be found because they haven't been built then. 

I specified sql-plugin-api and it passed. But that means we can only pre-download sql-plugin-api to speedup the workflow. Is that what was expected?

As I skimmed the log, it seems the cache cannot be used in following jobs. Or should we use `cache` action in a isolated job to pre-download the dependencies and store them in `.m2` repo? I'm not sure if the `.m2` repo can be shared between jobs.

---

> I specified sql-plugin-api and it passed. But that means we can only pre-download sql-plugin-api to speedup the workflow. Is that what was expected?

Missed your comment. Answer: it is in internal module we were just missing `-am`